src/platform/forms/components/review/PreSubmitSection.jsx

```diff
@@ -52,6 +58,7 @@ export function PreSubmitSection(props) {
           )}
         </div>
       )}
+      {saveLink}
```

Comment:
It seems like the `saveLink` prop is added to several components so that they can pass it to their children (and so on and so forth), just so that we can use it here. Is it possible for us to skip all those middle men?

Reply:
I tried passing down the `pageList` and `locationPathname` down through all the child components, which is essentially the same headache, but as luck would have it, it didn't seem to work. I don't know how to get the values from `pageList` and `route` other than passing it down - since those values don't live in the redux state.

Reply:
In the `propTypes` for this file I'm seeing:

- form
- formConfig
- showPresubmitError
- setPreSubmit

And the `<SaveFormLink>` in `RoutedSavableReviewPage` is taking the following props:

- locationPathname
- form
- formConfig
- user
- pageList
- showLoginModal
- saveAndRedirectToReturnUrl
- toggleLoginModal

`form` and `formConfig` we already have. We could probably get `locationPathname` directly from the url. I think `user` comes from redux, and it looks like `showLoginModal` does as well. If that's true, then the only unaccounted for props are:

- pageList
- saveAndRedirectToReturnUrl
- toggleLoginModal

So, if we could provide those additional props, would it make sense to construct the `saveLink` in this file rather than in one of the *-grandparents and pass it down?